src/components/app/AppLocalizationProvider.js

```diff
@@ -99,16 +99,69 @@ class AppLocalizationFetcher extends React.PureComponent<FetchProps> {
 
 type InitProps = {|
   +requestL10n: typeof requestL10n,
+  +requestedLocales: null | string[],
 |};
 
 /**
- * This component is responsible for initializing the locales.
+ * This component is responsible for initializing the locales as well as
+ * persisting the current locale to localStorage.
  */
 class AppLocalizationInit extends React.PureComponent<InitProps> {
   componentDidMount() {
     const { requestL10n } = this.props;
-    const locales = navigator.languages;
-    requestL10n(locales);
+    requestL10n(this._getPersistedLocale() ?? navigator.languages);
```

Comment:
Right now, when we persist something, we can no longer go back to using `navigator.languages`. Only way to undo that is to remove the local storage item, I guess. But I don't think this is something important for our users. It's more like a mental note for me, if I work on that in the future :)